server/controllers/finance/cashflow.js

```diff
@@ -0,0 +1,476 @@
+					'FROM `posting_journal` ' +
+					'WHERE `posting_journal`.`account_id` IN (?) AND `posting_journal`.`trans_date` >= ? AND `posting_journal`.`trans_date` <= ? ' +
+				') UNION (' +
+					'SELECT `general_ledger`.`project_id`, `general_ledger`.`uuid`, `general_ledger`.`record_uuid`, `general_ledger`.`trans_date`, `general_ledger`.`debit_equiv`, ' +
+						'`general_ledger`.`credit_equiv`, `general_ledger`.`debit`, `general_ledger`.`credit`, `general_ledger`.`account_id`, `general_ledger`.`entity_uuid`, ' +
+						'`general_ledger`.`currency_id`, general_ledger.trans_id, `general_ledger`.`description`, `general_ledger`.`comment`, `general_ledger`.`origin_id`, `general_ledger`.`user_id` ' +
+					'FROM `general_ledger` ' +
+					'WHERE `general_ledger`.`account_id` IN (?) AND `general_ledger`.`trans_date` >= ? AND `general_ledger`.`trans_date` <= ? ' +
+				')' +
+			') AS `t`, account AS a, user as u WHERE `t`.`account_id` = `a`.`id` AND  `t`.`user_id` = `u`.`id` ' +
+			'GROUP BY `t`.`trans_id` ;';
+
+	return db.exec(requette, [params.account_id, params.dateFrom, params.dateTo, params.account_id, params.dateFrom, params.dateTo]);
+}
+
+/**
+ * @function groupingIncomeExpenseByPeriod
+ * @description This function help to group incomes or expenses by period
+ */
+function groupingIncomeExpenseByPeriod(periodicFlows) {
+	var grouping = [];
+	periodicFlows.forEach(function (pf) {
+		var incomes, expenses;
+		incomes = pf.flows.filter(function (posting) {
+			return posting.debit_equiv > 0;
+		});
+		expenses = pf.flows.filter(function (posting) {
+			return posting.credit_equiv > 0;
+		});
+		grouping.push({ period: pf.period, incomes : incomes, expenses : expenses });
+	});
+	return grouping;
+}
+
+/**
+ * @function groupingByMonth
+ * @param {array} periods An array which contains all periods for the fiscal year
+ * @param {array} flows The result of `queryIncomeExpense` i.e. all incomes and expense
+ * @description This function help to group incomes or expenses by month
+ */
+function groupingByMonth(periods, flows) {
+	var grouping = [];
+	periods.forEach(function (p) {
+		var data = [];
+		flows.forEach(function (f) {
+			var trans_date = new Date(f.trans_date);
+			var start_date = new Date(p.start_date);
+			var end_date = new Date(p.end_date);
+			if (trans_date <= end_date && trans_date >= start_date) {
+				data.push(f);
+			}
+		});
+		grouping.push({ period: p, flows : data });
+	});
+	return grouping;
+}
+
+/**
+ * @function closingBalance
+ * @param {number} accountId An account for which we search to know the balance
+ * @param {date} periodStart The first period start of a given fiscal year (current fiscal year)
+ * @desc This function help us to get the balance at cloture for a set of accounts
+ */
+function closingBalance(accountId, periodStart) {
```

Comment:
I think you could probably get these values much more efficiently from the period_totals table.

Should the cash flow report include data from the `posting_journal`?